lib/utils/job-utils/catalog-searcher.js

```diff
@@ -71,70 +69,138 @@ function searchCatalogDataFactory(
+                controllerVendors[controllerId] = vendor.toLowerCase();
+            });
+            return controllerVendors;
+        });
+    }
+
+    /**
+     * Get extended driveId catalog
+     * @param {String} nodeId - node identifier
+     * @param {Object} driveIds - driveId list retrieved from driveId catalogs
+     * @param {Boolean} extendJbod - flag for if we should extend JBOD physical information
+     * @return {Promise} Drive catalogs extended with Megaraid information
+     */
+    function getDriveIdCatalogExt(nodeId, filter, extendJbod) {
+        return getDriveIdCatalog(nodeId, filter)
+        .then(function (driveIds) {
             // get virtualDisk data from megaraid-virtual-disks catalog
             var foundVdHasValue = _.find(driveIds, function (driveId) {
                 return !_.isEmpty(driveId.virtualDisk);
             });
 
-            if (!foundVdHasValue) {
-                return;
+            if (!foundVdHasValue && !extendJbod) {
+                // If none of drives are VDs, and extend JBOD is not required,
+                // it is not necessary to extend Megaraid information.
+                return Promise.resolve(driveIds);
             }
-            return waterline.catalogs.findMostRecent({
-                node: nodeId,
-                source: 'megaraid-virtual-disks'
-            })
-            .then(function (virtualDiskCatalog) {
-                if (!_.has(virtualDiskCatalog, 'data.Controllers[0]')) {
-                    return Promise.reject(
-                        new Error('Could not find megaraid-virtual-disks catalog data.'));
-                }
-                virtualDiskData = virtualDiskCatalog.data;
-            });
-        })
-        .tap(function () {
-            // get controller vendor information from megaraid-controllers
-            return waterline.catalogs.findMostRecent({
-                node: nodeId,
-                source: 'megaraid-controllers'
-            })
-            .then(function (controllerCatalog) {
-                if (!_.has(controllerCatalog, 'data.Controllers')) {
-                    return Promise.reject(
-                        new Error('Could not find megaraid-controllers catalog data.'));
-                }
-                var vendor, controllerId;
-                var controllerDataList = _.get(controllerCatalog, 'data.Controllers');
-                _.forEach(controllerDataList, function(data){
-                    vendor = _.get(data, '[Response Data][Scheduled Tasks].OEMID');
-                    controllerId = _.get(data, '[Command Status][Controller]');
-                    controllerVendors[controllerId] = vendor.toLowerCase();
+
+            return Promise.all([
+                foundVdHasValue ? getVirtualDiskCatalog(nodeId) : Promise.resolve(),
+                extendJbod ? getPhysicalDiskCatalog(nodeId) : Promise.resolve(),
+                getRaidControllerVendor(nodeId)
```

Comment:
you could add "_" in the beginning of the above three function names to signify they are used internally.